pkg/filters/proxy/pool.go

```diff
@@ -639,6 +641,24 @@ func (sp *ServerPool) newMetrics(name string) *metrics {
 				Buckets: prometheushelper.DefaultBodySizeBuckets(),
 			},
 			proxyLabels).MustCurryWith(commonLabels),
+		RequestBodySizePercentage: prometheushelper.NewSummary(
+			prometheus.SummaryOpts{
+				Name: "proxy_request_body_size_percentage",
+				Help: "a summary of the total size of the request.",
+				Objectives: map[float64]float64{
+					0.25: 20, 0.5: 20, 0.9: 20, 0.95: 20, 0.99: 20,
+				},
+			},
+			proxyLabels).MustCurryWith(commonLabels),
+		ResponseBodySizePercentage: prometheushelper.NewSummary(
+			prometheus.SummaryOpts{
+				Name: "proxy_response_body_size_percentage",
+				Help: "a summary of the total size of the response.",
+				Objectives: map[float64]float64{
```

Comment:
what's the usage of this map?
as floating point numbers often can not be represented precisely, we must be cautious about using them as keys of maps.